tracer/src/Datadog.Trace/TracerManagerFactory.cs

```diff
@@ -350,6 +351,15 @@ private static string GetApplicationName(ImmutableTracerSettings settings)
                     return serviceName;
                 }
 
+                if (ServerlessMiniAgent.GetIsGCPFunction() || ServerlessMiniAgent.GetIsAzureFunction())
+                {
+                    var functionName = ServerlessMiniAgent.GetGCPAzureFunctionName();
```

Comment:
For Azure, this is a duplicate of line 344. For consistency, I would have followed the same pattern ie:
- create GCPImmutableSettings (or rename then to HostingImmutableSettings if we want to share things between the 2?)
- add on line 348 the following

```
                if (settings.IsRunningInGCR)
                {
                    return settings.GCRMetadata.FunctionName;
                }
```

Reply:
I've created a GCPImmutableSettings, and moved the two GetIsXFunction() methods into the respective immutable settings files.